src/spark-core.js

```diff
@@ -143,7 +145,14 @@ var SparkCore = AmpersandState.extend({
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
       })
-      .then(this.credentials.logout.bind(this.credentials));
+      .then(function clearEncryptStore() {
+        this.encryption.keystore.clear();
```

Comment:
Clearing the keys return a promise but unfortunately that promise isn't propagated up by ampersand.